specs/electra/beacon-chain.md

```diff
@@ -1459,12 +1434,23 @@ def process_consolidation_request(
     if target_validator.exit_epoch != FAR_FUTURE_EPOCH:
         return
 
+    # Churn any target excess active balance of target and raise its max
+    if has_eth1_withdrawal_credential(target_validator):
+        state.validators[target_index].withdrawal_credentials = (
+            COMPOUNDING_WITHDRAWAL_PREFIX + target_validator.withdrawal_credentials[1:]
+        )
+        queue_excess_active_balance(state, target_index)
+
+    # Verify that source != target, so a consolidation cannot be used as an exit.
+    if source_index == target_index:
+        return
+
     # Initiate source validator exit and append pending consolidation
-    source_validator.exit_epoch = compute_consolidation_epoch_and_update_churn(
+    state.validators[source_index].exit_epoch = compute_consolidation_epoch_and_update_churn(
```

Comment:
I think in Python variable is assigned to a reference. So if you modify a variable that is a reference to an element in an array, the array element will be updated as well. 

I don't see why we would need to replace `source_validator` with `state.validators[source_index]` explicitly when updating. Unless I am missing something.

Reply:
Yes, you’re right, and the reason for doing so is the bug somewhere in one of the components that spec relies on, likely in remerklable. Consider the following test:
```python
@with_electra_and_later
@spec_state_test
def test_state_changes_overwritten(spec, state):
    validator_0 = state.validators[0]
    validator_1 = state.validators[1]
    
    validator_0.exit_epoch = spec.Epoch(0)
    validator_1.exit_epoch = spec.Epoch(1)

    assert state.validators[1].exit_epoch == spec.Epoch(1)
    assert state.validators[0].exit_epoch == spec.Epoch(0)
```
The second assert fails while it must not. This problem requires a separate work, and I decided to use a workaround for this spec change